src/commonMain/kotlin/software/momento/kotlin/sdk/auth/CredentialProvider.kt

```diff
@@ -0,0 +1,85 @@
+package software.momento.kotlin.sdk.auth
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+
+/**
+ * Contains the information required for a Momento client to connect to and authenticate with
+ * the Momento service.
+ */
+public data class CredentialProvider(
+    val controlEndpoint: String, val cacheEndpoint: String, val apiKey: String
+) {
+    public companion object {
+        //TODO: throw Momento exceptions when those are added
+
+        /**
+         * Creates a [CredentialProvider] from a Momento API key.
+         * @param apiKey The Momento API key to use for authentication.
+         * @param controlHost An optional override for the endpoint used for control operations.
+         * @param cacheHost An optional override for the endpoint used for cache operations.
+         */
+        public fun fromString(
+            apiKey: String, controlHost: String? = null, cacheHost: String? = null
+        ): CredentialProvider {
+            try {
+                val provider = try {
+                    processLegacyKey(apiKey)
```

Comment:
Since this is a new SDK, should we get rid of this concept altogether? Not a strong opinion though and also curious if the legacy ones add any value anymore